exchanges/ftx/ftx_wrapper.go

```diff
@@ -0,0 +1,833 @@
+			return nil, err
+		}
+	}
+	return ticker.GetTicker(f.Name, p, assetType)
+}
+
+// FetchTicker returns the ticker for a currency pair
+func (f *FTX) FetchTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	tickerNew, err := ticker.GetTicker(f.Name, p, assetType)
+	if err != nil {
+		return f.UpdateTicker(p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// FetchOrderbook returns orderbook base on the currency pair
+func (f *FTX) FetchOrderbook(currency currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	ob, err := orderbook.Get(f.Name, currency, assetType)
+	if err != nil {
+		return f.UpdateOrderbook(currency, assetType)
+	}
+	return ob, nil
+}
+
+// UpdateOrderbook updates and returns the orderbook for a currency pair
+func (f *FTX) UpdateOrderbook(p currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	orderBook := new(orderbook.Base)
+	tempResp, err := f.GetOrderbook(f.FormatExchangeCurrency(p, assetType).String(), 0)
+	if err != nil {
+		return orderBook, err
+	}
+	for x := range tempResp.Bids {
+		orderBook.Bids = append(orderBook.Bids, orderbook.Item{
+			Amount: tempResp.Bids[x].Size,
+			Price:  tempResp.Bids[x].Price})
+	}
+	for y := range tempResp.Asks {
+		orderBook.Asks = append(orderBook.Asks, orderbook.Item{
+			Amount: tempResp.Asks[y].Size,
+			Price:  tempResp.Asks[y].Price})
+	}
+	orderBook.Pair = p
+	orderBook.ExchangeName = f.Name
+	orderBook.AssetType = assetType
+	err = orderBook.Process()
+	if err != nil {
+		return orderBook, err
+	}
+	return orderbook.Get(f.Name, p, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (f *FTX) UpdateAccountInfo() (account.Holdings, error) {
+	var resp account.Holdings
+	data, err := f.GetBalances()
+	if err != nil {
+		return resp, err
+	}
+	var acc account.SubAccount
+	for key := range data.Result {
+		c := currency.NewCode(data.Result[key].Coin)
+		hold := data.Result[key].Total - data.Result[key].Free
+		total := data.Result[key].Total
+		acc.Currencies = append(acc.Currencies,
+			account.Balance{CurrencyName: c,
+				TotalValue: total,
+				Hold:       hold})
+	}
+	resp.Accounts = append(resp.Accounts, acc)
+	resp.Exchange = f.Name
+
+	err = account.Process(&resp)
+	if err != nil {
+		return account.Holdings{}, err
+	}
+
+	return resp, nil
+}
+
+// FetchAccountInfo retrieves balances for all enabled currencies
+func (f *FTX) FetchAccountInfo() (account.Holdings, error) {
+	acc, err := account.GetHoldings(f.Name)
+	if err != nil {
+		return f.UpdateAccountInfo()
+	}
+
+	return acc, nil
+}
+
+// GetFundingHistory returns funding history, deposits and
+// withdrawals
+func (f *FTX) GetFundingHistory() ([]exchange.FundHistory, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetExchangeHistory returns historic trade data since exchange opening.
+func (f *FTX) GetExchangeHistory(p currency.Pair, assetType asset.Item) ([]exchange.TradeHistory, error) {
+	return nil, common.ErrNotYetImplemented
```

Comment:
Please implement this

Reply:
also for every asset type the very first start date is different too. like some contracts only started in 2020 so thats another issue. and also have tried setting just the end date so could prepend all the data n go back in history till we stop gettin trades but they dont allow just end date either